server/src/main/java/org/apache/iotdb/db/query/dataset/NonAlignEngineDataSet.java

```diff
@@ -0,0 +1,354 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.query.dataset;
+
+import org.apache.iotdb.db.query.pool.QueryTaskPoolManager;
+import org.apache.iotdb.db.query.reader.ManagedSeriesReader;
+import org.apache.iotdb.db.tools.watermark.WatermarkEncoder;
+import org.apache.iotdb.service.rpc.thrift.TSQueryNonAlignDataSet;
+import org.apache.iotdb.tsfile.exception.write.UnSupportedDataTypeException;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.read.common.BatchData;
+import org.apache.iotdb.tsfile.read.common.Path;
+import org.apache.iotdb.tsfile.read.common.RowRecord;
+import org.apache.iotdb.tsfile.read.query.dataset.QueryDataSet;
+import org.apache.iotdb.tsfile.utils.Pair;
+import org.apache.iotdb.tsfile.utils.PublicBAOS;
+import org.apache.iotdb.tsfile.utils.ReadWriteIOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicIntegerArray;
+
+public class NonAlignEngineDataSet extends QueryDataSet {
+
+  private static class ReadTask implements Runnable {
+
+    private final ManagedSeriesReader reader;
+    private BlockingQueue<Pair<ByteBuffer, ByteBuffer>> blockingQueue;
+    private WatermarkEncoder encoder;
+    NonAlignEngineDataSet dataSet;
+    private int index;
+
+
+    public ReadTask(ManagedSeriesReader reader,
+                    BlockingQueue<Pair<ByteBuffer, ByteBuffer>> blockingQueue,
+                    WatermarkEncoder encoder, NonAlignEngineDataSet dataSet, int index) {
+      this.reader = reader;
+      this.blockingQueue = blockingQueue;
+      this.encoder = encoder;
+      this.dataSet = dataSet;
+      this.index = index;
+    }
+
+    @Override
+    public void run() {
+      PublicBAOS timeBAOS = new PublicBAOS();
+      PublicBAOS valueBAOS = new PublicBAOS();
+      try {
+        synchronized (reader) {
+          // if the task is submitted, there must be free space in the queue
+          // so here we don't need to check whether the queue has free space
+          // the reader has next batch
+          if ((dataSet.cachedBatchData[index] != null && dataSet.cachedBatchData[index].hasCurrent())
+                  || reader.hasNextBatch()) {
+            BatchData batchData;
+            if (dataSet.cachedBatchData[index] != null && dataSet.cachedBatchData[index].hasCurrent())
+              batchData = dataSet.cachedBatchData[index];
+            else
+              batchData = reader.nextBatch();
```

Comment:
Always use "{}" for code blocks (even if it only contains a single line), as we obey the google code-style.

Reply:
fixed